tests/test_cases/test_cocotb/test_deprecated.py

```diff
@@ -106,3 +109,21 @@ async def test_handle_compat_mapping(dut):
     with assert_deprecated():
         dut.fullname = "myfullname"
     assert dut.fullname == "myfullname"
+
+
+@cocotb.test()
+async def test_assigning_structure_deprecated(dut):
+    """signal <= ctypes.Structure assignment is deprecated"""
+
+    class Example(ctypes.Structure):
+        _fields_ = [
+            ("a", ctypes.c_byte)]
+
+    e = Example(a=0xCC)
```

Comment:
To make this test really check the recommended behavior is the same, you should use a larger ctypes structure that occupies multiple bytes (maybe more than 4 to be sure?)

Reply:
```suggestion
    class Example(ctypes.Structure):
        _fields_ = [
            ("b", ctypes.c_uint32), ("a", ctypes.c_byte)]

    e = Example(a=0xCC, b=0x12345678)
```
How wide is `stream_in_data`?

Reply:
8 bits

Reply:
There is `dut.stream_in_data_width`, but `BinaryValue` has issues with wide values. I could try it.